OptionSelector/OptionSelectorTableViewController.m

```diff
@@ -0,0 +1,205 @@
+//
+//  OptionSelectorTableViewController.m
+//  OptionSelector
+//
+//  Created by Mesfin Bekele Mekonnen on 8/10/15.
+//  Copyright (c) 2015 Mike Kavouras. All rights reserved.
+//
+
+#import "OptionSelectorTableViewController.h"
+#import "OptionSelectorDetailTableViewController.h"
+#import "OptionSelectorDetailTableViewControllerDelegate.h"
+#import "Type.h"
+
+@interface OptionSelectorTableViewController () <OptionSelectorDetailTableViewControllerDelegate>
+
+@property (nonatomic) NSMutableArray *array;
+@property (nonatomic) NSInteger index;
+
+@end
+
+@implementation OptionSelectorTableViewController
+
+- (IBAction)saveButtonTapped:(UIBarButtonItem *)sender {
+    
+    NSArray *indexesOfCells = [self.tableView visibleCells];
+    for (int i=0; i < indexesOfCells.count; i++) {
+        UITableViewCell *tmp = indexesOfCells[i];
+        if (i == 0) {
+            NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+            [defaults setObject:tmp.detailTextLabel.text forKey:@"cell1"];
+        }
+        else if(i == 1){
+            NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+            [defaults setObject:tmp.detailTextLabel.text forKey:@"cell2"];
+        }
+        else{
+            NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+            [defaults setObject:tmp.detailTextLabel.text forKey:@"cell3"];
+        }
+    }
+    
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.array = [[NSMutableArray alloc]init];
```

Comment:
`array` is a non-descriptive property name. Better would be `types`, or `categories`, or something that tells the programmer what sorts of things this array holds.